packages/hooks/src/check-update.js

```diff
@@ -0,0 +1,330 @@
+  }
+  return undefined;
+}
+
+/**
+ * Checks if the latest version is more recent than the current version.
+ * @param {string | undefined} current - Package version available in the project.
+ * @param {string | undefined} latest - Most up-to-date dependency version available.
+ * @returns {boolean} If the update will result in a breaking change.
+ */
+export function hasAvailableUpdates(current, latest) {
+  if (!current || !latest) {
+    return false;
+  }
+  const [currMajor, currMinor, currPatch] = current
+    .split('.')
+    .map((val) => Number(val));
+  const [targetMajor, targetMinor, targetPatch] = latest
+    .split('.')
+    .map((val) => Number(val));
+  if (targetMajor !== currMajor) {
+    return targetMajor > currMajor;
+  } if (targetMinor !== currMinor) {
+    return targetMinor > currMinor;
+  }
+  return targetPatch > currPatch;
+}
+
+/**
+ * Checks if updating a dependency to the latest version causes a breaking change.
+ * @param {string | undefined} current - Package version available in the project.
+ * @param {string | undefined} latest - Most up-to-date dependency version available.
+ * @returns {boolean} If the update will result in a breaking change.
+ */
+export function hasBreakingChange(current, latest) {
+  if (!current || !latest) {
+    return false;
+  }
+  const currMajor = current.split('.')[0];
+  const latestMajor = latest.split('.')[0];
+  return +latestMajor - +currMajor >= 1;
+}
+
+/**
+ * Wraps and parses the output of the exec() function
+ * @param {string} command - The command to soon be executed.
+ * @returns {Promise<string>} the output from the command.
+ */
+async function execWrapper(command) {
+  const { stdout } = await exec(command);
+  return stdout.trim();
+}
+
+/**
+ * Reads and parses a JSON file to return the contents.
+ * @param {string} filePath - The path of the file being read.
+ * @returns {Promise<object>} - The parsed file contents.
+ */
+async function getJSON(filePath) {
+  if (fs.existsSync(filePath)) {
+    return JSON.parse(fs.readFileSync(filePath, 'utf8'));
+  }
+  throw new Error(`Cannot find a file at path ${filePath}`);
+}
+
+/**
+ * Creates the check update response in the format expected by the CLI.
+ * @param {Record<string, ReleaseInfo>} versionMap - Information about packages and updates.
+ * @param {InaccessibleFile[]} inaccessibleFiles - Array of files that could not be read.
+ * @returns {UpdateInfo} Update information formatted in the expected manner.
+ */
+function createCheckUpdateResponse(versionMap, inaccessibleFiles) {
+  /** @type {string[]} */
+  const dependencyErrors = [];
+  const releases = Object.entries(versionMap).map(([dependency, version]) => {
+    if (version.error) {
+      dependencyErrors.push(dependency);
+    }
+    return version;
+  });
+  return {
+    name: 'the Slack SDK',
+    message: '',
+    releases,
+    url: 'https://api.slack.com/automation/changelog',
```

Comment:
Again, is this what we would want to point them to?

Reply:
Ah, thanks for the visual! That's helpful. I guess my question is will the `hooks` package update be present in the Changelog with everything else? If so, it makes sense. If not, we probably need to tweak the CLI logic at some point.

Reply:
+1 to future changes in the CLI since I don't think these `hooks` will be included in the changelog / follow the same release cycle. Maybe we do align these releases though?

For now I think this is alright to leave but will add it to the backlog!